covid_analyzer/utils.py

```
@@ -0,0 +1,157 @@
+import matplotlib.pyplot as plt
+from constants import (
+    COUNTRY_NAME_IDX_IN_COVID_STATS, TOTAL_PATIENTS, TOTAL_RECOVERED_PATIENTS, COUNTRY_NAME_IDX_IN_COVID_SAFETY_MEASURES,
+    SAFETY_MEASURES, IMPLEMENTED_SAFETY_MEASURES, TOTAL_DEATHS, EMPTY_STRING, TOTAL_CASES, ZERO, EFFICIENCY, SAFETY_MEASURE,
+    CASES_STATS
+    )
+
+
+def get_ratio_of_recovered_patients(covid_data, country: str) -> None: 
+    """Return the ratio of recovered patients of a country
+
+    Args:
+        country (str): country name (case sensitive)
+ 
+    Returns:
+        int: 0 if the country's name or data do not exist, otherwise returns the ratio
+    """
+    
+    for covid_cases_details_per_country in covid_data[CASES_STATS]:
+        if covid_cases_details_per_country[COUNTRY_NAME_IDX_IN_COVID_STATS] == country:
+            try:
+                recovered_ratio = round(
+                    int(covid_cases_details_per_country[TOTAL_RECOVERED_PATIENTS])
+                    / int(covid_cases_details_per_country[TOTAL_PATIENTS]), 2
+                )
+                print(f'Recovered/total ratio: {recovered_ratio}')
+                return None
+            except ValueError:
+                 print("Country data not available")
+    else:
+        print("Country data not available")
```

Is there any specific need to raise these errors twice?

---

Yes, Value error occurs when the country name available is correct but we don't have countries stats like (TOTAL_RECOVERED_PATIENTS)  in our data. The else block runs when we don't  have the country name provided as argument or it is mistyped.